deepspeed/runtime/zero/stage2.py

```diff
@@ -318,8 +325,8 @@ def __init__(self,
                 i].requires_grad = True  # keep this in case internal optimizer uses it
             param_group['params'] = [self.single_partition_of_fp32_groups[i]]
 
-            partition_size = len(self.fp16_groups_flat[i]) / dist.get_world_size(
-                group=self.real_dp_process_group[i])
+            #partition_size is same across all partitions
+            partition_size = data_parallel_partitions[0].numel()
```

Comment:
Can we leave this part as it was before as both are doing the same thing here? I do not see the benefit of the change but I think we should add an assertion partition_size % alignment == 0 here.

Reply:
len(self.fp16_groups_flat[i]) / world_size != data_parallel_partitions[0].numel()
They are not doing the same thing; The new code uses ceil() and therefore, can be len(self.fp16_groups_flat[i]) / world_size **+ 1** in some cases.

Reply:
I think in this case they will always be the same because  self.fp16_groups_flat[i].numel() % (world_size*nccl_alignment) == 0 due to the padding in flatten_dense_tensor_aligned.